presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/reloading/ReloadingResourceGroupConfigurationManager.java

```diff
@@ -244,88 +223,22 @@ public synchronized void load()
         }
     }
 
-    // Populate temporary data structures to build resource group specs and selectors from db
-    private synchronized void populateFromDbHelper(Map<Long, ResourceGroupSpecBuilder> recordMap,
-            Set<Long> rootGroupIds,
-            Map<Long, ResourceGroupIdTemplate> resourceGroupIdTemplateMap,
-            Map<Long, Set<Long>> subGroupIdsToBuild)
+    private synchronized Map<ResourceGroupIdTemplate, ResourceGroupSpec> buildResourceGroupSpecsMap(Map<ResourceGroupIdTemplate, ResourceGroupSpec> resourceGroupSpecs, List<ResourceGroupSpec> specsList, String path)
```

Comment:
instead of using `String path` we should use the structured object `Optional<ResourceGroupIdTemplate> parentId` - where this will be `Optional.empty()` for root groups